File: staircase.cpp
// first read the Readme file to have more clarity
#include<iostream>
using namespace std;

class staircase{
    public:
        staircase(int n){
            for(int i=1;i<n+1;i++){
                for(int k=1;k<=n-i;k++){
                    printf(" ");  
                }
                for(int j=i;j>0;j--){
                    printf("#");
                }
            printf("\n");
            }
        }
};

int main(){

    int n;
    cout<<"Enter the size of staircase(n) : ";
    cin>>n;
    staircase s(n);
    return 0;

}

File: staircase.py
#first read the Readme file to have more clarity.
def staircase(n):
    for i in range(1,n+1):
        for j in range(n,0,-1):
            if j>i :
                print(" ",end="")
            else:
                print("#",end="")
        print()

n = int(input("Enter the size of staircase(n) : "))
staircase(n)